src/Umbraco.Web.UI.Client/src/views/common/dashboard.controller.js

```diff
@@ -20,23 +22,38 @@ function DashboardController($scope, $routeParams, dashboardResource, localizati
     });
     
     dashboardResource.getDashboard($routeParams.section).then(function(tabs){
-        $scope.dashboard.tabs = tabs;
-        
-        // set first tab to active
-        if($scope.dashboard.tabs && $scope.dashboard.tabs.length > 0) {
-            $scope.dashboard.tabs[0].active = true;
-        }
-
+        $scope.dashboard.tabs = tabs;        
+        initActiveTab();
         $scope.page.loading = false;
     });
 
     $scope.changeTab = function(tab) {
         $scope.dashboard.tabs.forEach(function(tab) {
             tab.active = false;
         });
+
         tab.active = true;
+
+        $location.search(DASHBOARD_QUERY_PARAM, tab.alias);
     };
 
+    function initActiveTab () {
+        if (!$scope.dashboard.tabs) {
```

Comment:
Before there was a check for $scope.dashboard.tabs.length > 0, I think we need to add that again then. to avoid eventual issue on line 51.

Reply:
Good catch. I have updated the PR.